src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -337,7 +339,10 @@ static int ssend_ack_event(struct fi_cq_tagged_entry *wc, MPIR_Request * sreq)
 
 static uintptr_t recv_rbase(MPIDI_OFI_huge_recv_t * recv_elem)
 {
-    if (MPIDI_OFI_ENABLE_MR_SCALABLE) {
+    int ofi_version = MPIDI_OFI_get_ofi_version();
+
+    if ((ofi_version < FI_VERSION(1, 5) && MPIDI_OFI_ENABLE_MR_SCALABLE) ||
+        (ofi_version >= FI_VERSION(1, 5) && !MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS)) {
```

Comment:
Similarly I think we should consolidate this condition into a variable with more direct meaningful names that is set during init time.